cpp/src/parquet/file_deserialize_test.cc

```diff
@@ -196,6 +197,9 @@ class TestPageSerde : public ::testing::Test {
                         bool verification_checksum, bool has_dictionary = false,
                         bool write_data_page_v2 = false);
 
+  void TestPageCompressionRoundTrip(int num_pages,
+                                    const std::function<int(int)>& getPageSize);
```

Comment:
Why not allow simply passing a vector of page sizes?
```suggestion
  void TestPageCompressionRoundTrip(const std::vector<int32_t>& page_sizes);
```